src/Testing/Http/HttpRouterTester.php

```diff
@@ -47,40 +47,8 @@ public function sendRequest(Request $request): TestResponseHelper
         /** @var Router $router */
         $router = $this->container->get(Router::class);
 
-        // Let's check whether the current request matches a route
-        $matchedRoute = $router->matchRoute($request);
-
-        // If not, there's nothing left to do, we can't send this request
-        if (! $matchedRoute) {
-            throw new Exception("No matching route found for {$request->getMethod()->value} {$request->getPath()}");
-        }
-
-        // If we have a match, let's find out if our input request data matches what the route's action needs
-        $requestClass = $request::class;
-
-        // We'll loop over all the handler's parameters
-        foreach ($matchedRoute->route->handler->getParameters() as $parameter) {
-            // TODO: support unions
-
-            // If the parameter's type is an instance of Request…
-            if (is_a($parameter->getType()->getName(), Request::class, true)) {
-                // We'll use that specific request class
-                $requestClass = $parameter->getType()->getName();
-
-                break;
-            }
-        }
-
-        // We map the original request we got into this method to the right request class
-        $request = map($request)->to($requestClass);
-
-        // Finally, we register this newly created request object in the container
-        // This makes it so that RequestInitializer is bypassed entirely when the controller action needs the request class
-        // Making it so that we don't need to set any $_SERVER variables and stuff like that
-        $this->container->singleton(Request::class, fn () => $request);
-        $this->container->singleton($request::class, fn () => $request);
-
-        // Ok, now finally for real, we dispatch our request and return the response
-        return new TestResponseHelper($router->dispatch($request));
+        return new TestResponseHelper(
+            $router->dispatch(map($request)->to(PsrRequest::class))
```

Comment:
Why not just pass a PSR request to this method to begin with?

Also, don't we need to setup globals?

Reply:
Well my idea was that it's pretty convenient being able to call `$this->http->sendRequest(new MyRequest)` from within tests. You don't want to deal with PSRs there either. 

That's why I plan on removing the `request` function, in favor of this approach